omnigibson/sensors/vision_sensor.py

```diff
@@ -258,6 +296,39 @@ def add_modality(self, modality):
         # We also need to initialize this new modality
         if should_initialize:
             self.initialize_sensors(names=modality)
+    
+    def remove_modality(self, modality):
+        # Check if we don't have this modality (if not, no need to remove it explicitly)
+        should_remove = modality in self._modalities
+
+        # Run super
+        super().remove_modality(modality=modality)
```

Comment:
what exactly is the difference between remove_modality and remove_modality_from_backend?

Reply:
`remove_modality` handles everything we do internally. `remove_modality_from_backend` detaches the annotator corresponding to this modality from replicator.